<p> 1) Importe os dados e verifique as primeiras linhas do dataset. Verifique também as últimas linhas do dataset.</p>

In [1]:
import pandas as pd
df= pd.read_csv('Insurance.csv')
#verificando as primeiras linhas do dataset

df.head(5)

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [2]:
df.tail(10)

age     sex     bmi  children smoker     region      charges
1328   23  female  24.225         2     no  northeast  22395.74424
1329   52    male  38.600         2     no  southwest  10325.20600
1330   57  female  25.740         2     no  southeast  12629.16560
1331   23  female  33.400         0     no  southwest  10795.93733
1332   52  female  44.700         3     no  southwest  11411.68500
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

<p>2) Verifique se existe algum valor missing no dataset.</p>

In [3]:
#metodo 1 
#para verificar se existe algum valor missing no dataset
#não contem nulos

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [4]:
#metodo 2
#verificando valores nulos por coluna
valores_missing=df.isnull().sum()
print(valores_missing)

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64


<p>3) Quanto de memória o dataset está consumindo? </p>

<p> <b>resposta:</b> uma forma de verificar quanto de memória o dataset está consumindo também é utilizando a função df.info(). </p>

<p> Mas se você quiser uma visão mais detalhada use 'df.memory_usage() </p>

In [5]:
memoria_coluna= df.memory_usage(deep=True)
print(memoria_coluna)

Index         128
age         10704
sex         82942
bmi         10704
children    10704
smoker      79216
region      88308
charges     10704
dtype: int64


In [6]:
memoria_total=memoria_coluna.sum()
print(memoria_total)

293410


 <p> <b> exercício 4) </b> Gere as estatísticas descritivas das colunas numéricas do dataset. Qual a mediana da coluna BMI? E a média? Qual desses valores é maior? O que isso significa? Caso não saiba o que isso significa, passe os 2 valores ao chatGPT e pergunte a ele o que significa a média ser maior ou menor que a mediana </p>

In [10]:
df['bmi'].mean() #media da coluna



30.66339686098655

In [11]:
df['bmi'].median() #mediana

30.4

<p> A mediana indica que metade dos dados está dentro desse intervalo de valores, nesse caso de 30.4. E a média, indica que o número médio, é simplesmente 30.4. A média ser maior que a mediana implica que existe uma desigualdade, pois se a metade da distribuição contém até 30.4. E a média ainda ser 30.6, pode ser indícios de outliers que estão puxando a média para cima. Indicando que a distribuição está inclinada para direita. A média é bastante influenciada por grandes números </p>

<p> <b> 5) </b> Qual a proporção de fumantes no nosso dataset? E de não fumantes? </p>

In [14]:
df.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [25]:
#numero total
total_pessoas=df['smoker'].count()

#numero total de fumantes
total_fumantes=(df['smoker']=='yes').sum()

#numero total de não fumantes
total_nfumantes=(df['smoker']=='no').sum()

print(f'O total de fumantes é de: {total_fumantes}')
print(f'O total de não fumantes é de: {total_nfumantes}')
print(f' E o total de pessoas é de: {total_pessoas}')

O total de fumantes é de: 274
O total de não fumantes é de: 1064
 E o total de pessoas é de: 1338
